File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/components/parts/validators/MaxCharsValidator.java
package cz.cvut.fel.matyapav.afandroid.components.parts.validators;

import android.widget.EditText;

import com.tomscz.afrest.rest.dto.AFValidationRule;

import cz.cvut.fel.matyapav.afandroid.components.parts.AFField;
import cz.cvut.fel.matyapav.afandroid.utils.Localization;
import cz.cvut.fel.matyapav.afandroid.utils.Utils;

/**
 * Validates MAXLENGTH rule, which checks if number of chars is less or equal than value specified in rule.
 *
 * @author Pavel Matyáš (matyapav@fel.cvut.cz)
 *
 * @since 1.0.0.
 */
public class MaxCharsValidator implements AFValidator {

    @Override
    public boolean validate(AFField field, StringBuilder errorMsgs, AFValidationRule rule) {
        boolean validationIsFine = true;
        if(Utils.isFieldWritable(field.getFieldInfo().getWidgetType())){
            EditText textfield = (EditText) field.getFieldView();
            if (textfield.getText() != null && textfield.getText().toString().length() > Integer.parseInt(rule.getValue())) {
                validationIsFine = false;
            }
        }
        if(!validationIsFine){
            errorMsgs.append(Localization.translate("validation.maxchars"));
        }
        return true;
    }
}

File: AFMobile/AFWinPhone/AFWinPhone/components/parts/validators/MaxCharsValidator.cs
﻿using AFWinPhone.enums;
using AFWinPhone.utils;
using System;
using System.Text;
using Windows.UI.Xaml.Controls;

namespace AFWinPhone.components.parts.validators
{
    /// <summary>
    /// Validates MAXLENGTH rule, which checks if number of chars is less or equal than value specified in rule.
    /// </summary>
    class MaxCharsValidator : AFValidator
    {
        public bool validate(AFField field, StringBuilder errorMsgs, AFValidationRule rule)
        {
            bool validationIsFine = true;
            if (Utils.IsFieldWritable(field.getFieldInfo().getWidgetType()))
            {
                TextBox textfield = (TextBox)field.getFieldView();
                if (textfield.Text != null && textfield.Text.Length > Convert.ToInt32(rule.getValue()))
                {
                    validationIsFine = false;
                }
            }
            if (field.getFieldInfo().getWidgetType().Equals(SupportedWidgets.PASSWORD))
            {
                PasswordBox textfield = (PasswordBox)field.getFieldView();
                if (textfield.Password != null && textfield.Password.Length > Convert.ToInt32(rule.getValue()))
                {
                    validationIsFine = false;
                }
            }
            if (!validationIsFine)
            {
                errorMsgs.Append(Localization.translate("validation.maxchars"));
            }
            return validationIsFine;
        }
    }
}
